src/loop_closure_assistant.cpp

```diff
@@ -136,12 +142,16 @@ void LoopClosureAssistant::processInteractiveFeedback(const
     tf2::convert(transform, msg.transform);
     msg.child_frame_id = "scan_visualization";
     msg.header.frame_id = feedback->header.frame_id;
-    msg.header.stamp = node_->now();
+    msg.header.stamp = clock_->now();
     tfB_->sendTransform(msg);
 
-    scan.header.frame_id = "scan_visualization";
-    scan.header.stamp = node_->now();
-    scan_publisher_->publish(scan);
+    if (is_activated_) {
+      // get scan and publish
```

Comment:
Its easier to just reset and re-create the dynamic feedback object on activation / deactivation so that it doesn't exist even if we're not in the active state. Then there's no bool to track

Reply:
[240b36da8d981d0807f84ca975db992236f3e598]

I see. I reverted loop_closure_assistant. Now it is reset and re-create on `on_activate/deactivate`.  

https://github.com/SteveMacenski/slam_toolbox/blob/dbdf76009596ea0ba572b03151dc1e87bf594878/src/slam_toolbox_common.cpp#L76-L87

https://github.com/SteveMacenski/slam_toolbox/blob/dbdf76009596ea0ba572b03151dc1e87bf594878/src/slam_toolbox_common.cpp#L106-L117